File: web/projects/portal/src/app/binding/event/convert-chart-ref-event.ts
/*
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright © 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import { ViewsheetEvent } from "../../common/viewsheet-client/index";
import { ChartBindingModel } from "../data/chart/chart-binding-model";
import { Tool } from "../../../../../shared/util/tool";

/**
 * Event for common parameters for ConvertChartRefEvent
 */
export class ConvertChartRefEvent implements ViewsheetEvent {
   /**
    * Creates a new instance of <tt>ConvertChartRefEvent</tt>
    *
    * @param name the name of the chart
    * @param refName the name of the ref wanted to convert
    * @param changeType the conversion type
    */
   constructor(public name: string, public refNames: string[], public changeType: number,
      public table: string, public confirmed: boolean, binding: ChartBindingModel)
   {
      this.binding = Tool.shallowClone(binding);
      this.binding.availableFields = [];
   }

   binding: any;
}

File: core/src/main/java/inetsoft/web/binding/event/ConvertChartRefEvent.java
/*
 * This file is part of StyleBI.
 * Copyright (C) 2024  InetSoft Technology
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package inetsoft.web.binding.event;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import com.fasterxml.jackson.databind.annotation.JsonSerialize;
import inetsoft.web.binding.model.ChartBindingModel;
import inetsoft.web.viewsheet.event.ViewsheetEvent;
import org.immutables.value.Value;

/**
 * Class that encapsulates the parameters for convert chart ref event.
 *
 * @since 12.3
 */
@Value.Immutable
@JsonSerialize(as = ImmutableConvertChartRefEvent.class)
@JsonDeserialize(as = ImmutableConvertChartRefEvent.class)
public interface ConvertChartRefEvent extends ViewsheetEvent {
   /**
    * Get the name of the ref wanted to convert.
    * @return the name of the ref wanted to convert.
    */
   String[] refNames();

   /**
    * Get the conversion type.
    * @return the conversion type.
    */
   int changeType();

   /**
    * Get the table name.
    * @return table name.
    */
   String table();

   /**
    * @return the chart binding model
    */
   ChartBindingModel binding();
}
